diff -rc turtlebot3_manipulation_simulations/.git open_manipulator_with_tb3_simulations/.git
*** turtlebot3_manipulation_simulations/.git	2020-08-20 02:44:27.641500430 +0000
--- open_manipulator_with_tb3_simulations/.git	2020-08-20 02:44:27.545503322 +0000
***************
*** 1 ****
! gitdir: /home/ubuntu/environment/robomaker_turtlebot3_teleop/.git/modules/turtlebot3_teleop_simulation_ws/modules/src/turtlebot3_manipulation_simulations
--- 1 ----
! gitdir: /home/ubuntu/environment/robomaker_turtlebot3_teleop/.git/modules/turtlebot3_teleop_simulation_ws/modules/src/open_manipulator_with_tb3_simulations
Only in turtlebot3_manipulation_simulations: .gitignore
diff -rc turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/launch/empty_world.launch open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/launch/empty_world.launch
*** turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/launch/empty_world.launch	2020-08-20 02:44:45.876950721 +0000
--- open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/launch/empty_world.launch	2020-08-20 02:44:27.557502960 +0000
***************
*** 3,9 ****
    <arg name="use_robot_name" default="om_with_tb3"/>
  
    <!-- These are the arguments you can pass this launch file, for example paused:=true -->
!   <arg name="paused" default="true"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
--- 3,9 ----
    <arg name="use_robot_name" default="om_with_tb3"/>
  
    <!-- These are the arguments you can pass this launch file, for example paused:=true -->
!   <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
diff -rc turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/launch/rooms.launch open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/launch/rooms.launch
*** turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/launch/rooms.launch	2020-08-20 02:44:45.876950721 +0000
--- open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/launch/rooms.launch	2020-08-20 02:44:27.557502960 +0000
***************
*** 3,9 ****
    <arg name="use_robot_name" default="om_with_tb3"/>
  
    <!-- These are the arguments you can pass this launch file, for example paused:=true -->
!   <arg name="paused" default="true"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
--- 3,9 ----
    <arg name="use_robot_name" default="om_with_tb3"/>
  
    <!-- These are the arguments you can pass this launch file, for example paused:=true -->
!   <arg name="paused" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
***************
*** 39,42 ****
--- 39,52 ----
    <include file="$(find open_manipulator_with_tb3_gazebo)/launch/joint_controller.launch">
      <arg name="use_robot_name" value="$(arg use_robot_name)"/>
    </include>
+   
+   
+   <!-- added for work arm -->
+   <node name="gazebo_bridge" pkg="open_manipulator_with_tb3_tools" type="open_manipulator_with_tb3_gazebo_bridge" output="screen" ns="$(arg use_robot_name)"/>
+   <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
+     <rosparam param="source_list" subst_value="true">["$(arg use_robot_name)/joint_states"]</rosparam>
+       <param name="robot_description"
+       command="$(find xacro)/xacro --inorder '$(find open_manipulator_with_tb3_description)/urdf/open_manipulator_with_tb3_$(arg model).urdf.xacro'"/>
+   </node>
+ 
  </launch>
diff -rc turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/worlds/turtlebot3_rooms.world open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/worlds/turtlebot3_rooms.world
*** turtlebot3_manipulation_simulations/open_manipulator_with_tb3_gazebo/worlds/turtlebot3_rooms.world	2020-08-20 02:44:45.892950238 +0000
--- open_manipulator_with_tb3_simulations/open_manipulator_with_tb3_gazebo/worlds/turtlebot3_rooms.world	2020-08-20 02:44:27.573502478 +0000
***************
*** 49,54 ****
          <view_controller>orbit</view_controller>
        </camera>
      </gui>
- 
    </world>
  </sdf>
--- 49,53 ----
Only in turtlebot3_manipulation_simulations: .travis.yml
